app/src/main/java/com/android/sample/model/workout/WorkoutRepositoryFirestore.kt

```diff
@@ -121,8 +152,33 @@ class WorkoutRepositoryFirestore<T : Workout>(
         .collection(documentName)
         .document(id)
         .delete()
-        .addOnSuccessListener { onSuccess() }
         .addOnFailureListener { e -> onFailure(e) }
+
+    db.collection(mainDocumentName)
+        .document(id)
+        .delete()
+        .addOnFailureListener { e -> onFailure(e) }
+        .addOnSuccessListener { onSuccess() }
+  }
+
+  private fun documentSnapshotToObjectIds(doc: DocumentSnapshot): String? {
+    if (!doc.exists()) {
+      Log.e("DEB", "The document does not exist")
+      return null
+    }
+    return try {
+      val json = moshi.adapter(Map::class.java).toJson(doc.data)
+      adapterWorkoutID.fromJson(json)?.workoutid
+    } catch (e: JsonDataException) {
+      Log.e("Moshi", "Data id error JSON : ${e.message}")
+      throw e
+    } catch (e: JsonEncodingException) {
+      Log.e("Moshi", "Encoding id error JSON : ${e.message}")
+      throw e
+    } catch (e: Exception) {
+      Log.e("Moshi", "Conversion id error : ${e.message}")
+      throw e
+    }
   }
```

Comment:
I noticed that your methods documentSnapshotToObject and documentSnapshotToObjectIds are pretty much the same. Wouldn't it be better to avoid duplicating code lines and merge the two ?

Reply:
Done.